org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -431,11 +434,13 @@ public static IDocument getDocument(@Nullable IResource resource) {
 
 		if (document == null && resource.getType() == IResource.FILE) {
 			ITextFileBufferManager bufferManager = FileBuffers.getTextFileBufferManager();
+			if (bufferManager == null) 
+				return null;
 			try {
 				bufferManager.connect(resource.getFullPath(), LocationKind.IFILE, new NullProgressMonitor());
```

Comment:
Error during test thrown at connect, this line only one that was changed from the stack.

org.eclipse.core.runtime.CoreException: Cannot determine URI for '<file>'.
	at org.eclipse.core.internal.filebuffers.ResourceFileBuffer.create(ResourceFileBuffer.java:239)
	at org.eclipse.core.internal.filebuffers.TextFileBufferManager.connect(TextFileBufferManager.java:115)
	**at org.eclipse.lsp4e.LSPEclipseUtils.getDocument(LSPEclipseUtils.java:440)**
	at org.eclipse.lsp4e.LSPEclipseUtils.getDocument(LSPEclipseUtils.java:636)
	at org.eclipse.lsp4e.LSPEclipseUtils.getDocument(LSPEclipseUtils.java:616)
	at org.eclipse.lsp4e.outline.CNFOutlinePage.<init>(CNFOutlinePage.java:82)